lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -244,7 +244,7 @@ def formatted_message_and_backtrace(colorizer)
         end
 
         def exception_backtrace
-          exception.backtrace || []
+          RSpec.configuration.backtrace_formatter.format_backtrace(exception.backtrace) || []
```

Comment:
Updated cucumber feature again to have more clear example
@JonRowe 
> The snippet shows the code that triggered the failure

No its not completly true. Like in simple example
`it{ expect(true).to be_falsey }`
following your statement it should show `DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }` as snippet, because this line trigger failure, but it shows `it { expect(true).to be_falsey }` instead and to do so it filter backtrace here https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/formatters/exception_presenter.rb#L228-L234
Yes, it should not show rspec internal code, but then why it's not allowing filterring backtrace lines, to take a snippet, from user's loaded files?